src/DetectionConfig.php

```diff
@@ -0,0 +1,245 @@
+<?php declare(strict_types = 1);
+
+namespace ShipMonk\NameCollision;
+
+use JsonException;
+use LogicException;
+use Nette\Schema\Elements\Structure;
+use Nette\Schema\Expect;
+use Nette\Schema\Processor as SchemaProcessor;
+use Nette\Schema\ValidationException;
+use ShipMonk\NameCollision\Exception\InvalidConfigException;
+use function array_map;
+use function dirname;
+use function extension_loaded;
+use function file_get_contents;
+use function is_dir;
+use function is_file;
+use function is_readable;
+use function json_decode;
+use function json_last_error;
+use function json_last_error_msg;
+use function realpath;
+use const DIRECTORY_SEPARATOR;
+use const JSON_ERROR_NONE;
+use const JSON_PRESERVE_ZERO_FRACTION;
+
+class DetectionConfig
+{
+
+    /**
+     * @var list<string>
+     */
+    private $scanPaths;
+
+    /**
+     * @var list<string>
+     */
+    private $excludePaths;
+
+    /**
+     * @var list<string>
+     */
+    private $fileExtensions;
+
+    /**
+     * @var string
+     */
+    private $currentDirectory;
+
+    /**
+     * @var bool
+     */
+    private $ignoreParseFailures;
+
+    /**
+     * @param list<string> $scanPaths Absolute paths
+     * @param list<string> $excludePaths Absolute paths
+     * @param list<string> $fileExtensions
+     * @internal only for tests
+     */
+    public function __construct(
+        array $scanPaths,
+        array $excludePaths,
+        array $fileExtensions,
+        string $currentDirectory,
+        bool $ignoreParseFailures = false
+    )
+    {
+        $normalizePath = static function (string $path): string {
+            if (!is_dir($path) && !is_file($path)) {
+                throw new LogicException("Expected absolute path of existing file or dir, '$path' found.");
+            }
+
+            $absoluteRealPath = realpath($path);
+
+            if ($absoluteRealPath === false) {
+                throw new LogicException("Unable to realpath \"$path\" even though it is existing file or dir");
+            }
+
+            return $absoluteRealPath;
+        };
+
+        $this->scanPaths = array_map($normalizePath, $scanPaths);
+        $this->excludePaths = array_map($normalizePath, $excludePaths);
+        $this->currentDirectory = $currentDirectory;
```

Comment:
Current DIR is not normalized so I suppose it should be validated that it's an absolute path.